test/state/system_contracts.cpp

```diff
@@ -18,10 +18,13 @@ struct SystemContract
     evmc_revision since = EVMC_MAX_REVISION;  ///< EVM revision in which added.
     address addr;                             ///< Address of the system contract.
     GetInputFn* get_input = nullptr;          ///< How to get the input for the system call.
+    /// Type of requests returned by block end system call.
+    /// Ignored for block start system contracts.
+    Requests::Type request_type = Requests::Type::deposit;
```

Comment:
This abstraction is quite weak because we have disjoint set of objects with `get_input` and `requires_type`. Fine for me for now though.